include/sparrow/arrow_interface/arrow_schema.hpp

```diff
@@ -0,0 +1,143 @@
+
+#include "arrow_array_schema_utils.hpp"
+
+
+namespace sparrow
+{
+    /**
+     * Creates an ArrowSchema.
+     *
+     * @tparam C Value, reference or rvalue of std::vector<arrow_schema_shared_ptr>
+     * @tparam D Value, reference or rvalue of arrow_schema_shared_ptr
+     * @param format A mandatory, null-terminated, UTF8-encoded string describing the data type. If the data
+     *               type is nested, child types are not encoded here but in the ArrowSchema.children
+     *               structures.
+     * @param name An optional, null-terminated, UTF8-encoded string of the field or array name.
+     *             This is mainly used to reconstruct child fields of nested types.
+     * @param metadata An optional, binary string describing the type’s metadata. If the data type
+     *                 is nested, the metadata for child types are not encoded here but in the
+     * ArrowSchema.children structures.
+     * @param flags A bitfield of flags enriching the type description. Its value is computed by OR’ing
+     *              together the flag values.
+     * @param children An optional, std::vector<arrow_schema_shared_ptr>. Children pointers must not be null.
+     * @param dictionary An optional arrow_schema_shared_ptr. Must be present if the ArrowSchema represents a
+     * dictionary-encoded type. Must be nullptr otherwise.
+     * @return The created ArrowSchema unique pointer.
+     */
+    template <class F, class N, class M, class C, class D>
+        requires std::constructible_from<arrow_schema_private_data::FormatType, F>
+                 && std::constructible_from<arrow_schema_private_data::NameType, N>
+                 && std::constructible_from<arrow_schema_private_data::MetadataType, M>
+                 && std::constructible_from<arrow_schema_private_data::ChildrenType, C>
+                 && std::constructible_from<arrow_schema_private_data::DictionaryType, D>
+    [[nodiscard]] arrow_schema_unique_ptr
+    make_arrow_schema_unique_ptr(F format, N name, M metadata, std::optional<ArrowFlag> flags, C children, D dictionary);
+
+    template <class F, class N, class M, class C, class D>
+        requires std::constructible_from<arrow_schema_private_data::FormatType, F>
+                 && std::constructible_from<arrow_schema_private_data::NameType, N>
+                 && std::constructible_from<arrow_schema_private_data::MetadataType, M>
+                 && std::constructible_from<arrow_schema_private_data::ChildrenType, C>
+                 && std::constructible_from<arrow_schema_private_data::DictionaryType, D>
+    [[nodiscard]] arrow_schema_unique_ptr
+    make_arrow_schema_unique_ptr(F format, N name, M metadata, std::optional<ArrowFlag> flags, C children, D dictionary);
+
+    /**
+     * Release function to use for the `ArrowSchema.release` member.
+     */
+    void release_arrow_schema(ArrowSchema* schema);
+
+    /**
+     * Creates a default ArrowSchema unique pointer.
+     *
```

Comment:
Clarification of what are the default values would be helpful here.

Reply:
Ok done